ecogvis/functions/misc_dialogs.py

```diff
@@ -1541,9 +1598,1119 @@ def paint(self):
+        self.signal_stim_filt = speakerFilt
+        self.signal_resp_filt = micFilt
+
+        self.draw_scene()
+
+
+    def run_detection(self):
+        self.set_detect_interval()
+        self.disable_all()
+        self.plotTitle.setText(
+            f'Running event detection {self.detectStartTime}-'
+            f'{self.detectStopTime}s. Please wait...')
+        self.thread = EventDetectionFunction(
+            speaker_data=self.source_stim,
+            mic_data=self.source_resp,
+            interval=[self.detectStartBin, self.detectStopBin + 1],
+            dfact=self.fs / float(self.qline3.text()),
+            smooth_width=float(self.qline4.text()),
+            speaker_threshold=float(self.qline5.text()),
+            mic_threshold=float(self.qline6.text())
+        )
+        self.thread.finished.connect(lambda: self.out_close(1))
+        self.thread.start()
+
+    def disable_all(self):
+        self.win.clear()
+        self.push0_0.setEnabled(False)
+        self.push0_1.setEnabled(False)
+        self.push0_2.setEnabled(False)
+        self.combo0.setEnabled(False)
+        self.combo1.setEnabled(False)
+        self.qline1.setEnabled(False)
+        self.qline2.setEnabled(False)
+        self.qline3.setEnabled(False)
+        self.qline4.setEnabled(False)
+        self.qline5.setEnabled(False)
+        self.qline6.setEnabled(False)
+        self.qline7.setEnabled(False)
+        self.qline8.setEnabled(False)
+        self.push1_0.setEnabled(False)
+        self.push2_0.setEnabled(False)
+
+    def out_close(self, val):
+        self.value = val
+        if val == 1:  # finished auto-detection of events
+            self.stimTimes = self.thread.stimTimes
+            self.respTimes = self.thread.respTimes
+            fname = self.parent.model.fullpath
+            with NWBHDF5IO(fname, 'r+', load_namespaces=True) as io:
+                nwb = io.read()
+
+                # Speaker stimuli times
+                ti_stim = TimeIntervals(name='TimeIntervals_speaker')
+                times = self.stimTimes.reshape((-1, 2)).astype('float')
+                for start, stop in times:
+                    ti_stim.add_interval(start, stop)
+
+                # Microphone responses times
+                ti_resp = TimeIntervals(name='TimeIntervals_mic')
+                times = self.respTimes.reshape((-1, 2)).astype('float')
+                for start, stop in times:
+                    ti_resp.add_interval(start, stop)
+
+                # Add both to file
+                nwb.add_time_intervals(ti_stim)
+                nwb.add_time_intervals(ti_resp)
+                # Write file
+                io.write(nwb)
+            self.parent.model.refresh_file()
+            self.parent.model.SpeakerAndMicIntervalAdd()
+            self.parent.model.refreshScreen()
+        self.accept()
+
+
+# Runs 'detect_events' function, useful to wait for thread -----------------
+class EventDetectionFunction(QtCore.QThread):
```

Comment:
this has been moved out of here as well

Reply:
OK, I see that they have made changes to these functions. I rather we keep their changes and discard mine (which are just moving these classes to their own module)